### DISCUSSION TITLE
Granularity

### DISCUSSION THREAD

**Discussion by bitnom** [zauberzeug/nicegui#2398] [Q&A] [answered]
### Question

I'm thinking about using this thing. Will it allow me to get down & dirty if I need to? I'll intentionally leave that question vague, yet hope for a detailed response.

**Comment by falkoschindler** [comment, marked as answer]
Yes, NiceGUI allows you to fine-tune your UI using the underlying Quasar and Tailwind frameworks, with custom CSS or HTML code and custom JavaScript commands and events. You can also create your own custom components.

The ["examples" section](https://nicegui.io/#examples) gives a good impression of what is possible beyond the regular UI elements demonstrated in the documentation.
